lib/datadog/appsec/configuration.rb

```diff
@@ -40,6 +40,10 @@ def ip_denylist=(value)
           options[:ip_denylist] = value
```

Comment:
Wondering if we should rather resolve to a default here:
```suggestion
          options[:ip_denylist] = value || []
```

Reply:
Internally conflicted about allowing `nil` to mean "unset, use default" vs having a consistent type and eschew downstream `nil` checks.

This ties in with the whole configuration redesign, and I feel like the suggestion here is safer for now instead of littering the code with default values and nil checks at every call site.